src/main.rs

```diff
@@ -159,32 +162,36 @@ struct Opts {
     script: Option<PathBuf>,
 }
 
+pub type GitDate = Date<Utc>;
+
 #[derive(Clone, Debug)]
 enum Bound {
     Commit(String),
```

Comment:
And maintaining a connection between the `Commit` struct and the `Bound::Commit` enum or a link back to the GH data pulled into the `Commit` struct?  IMO it would be helpful to have the commit message string for the final report, particularly when this lands on a rollup PR commit where rolled up PR sha strings are included in the git commit message.

Reply:
hmm.  My interpretation is that `Bound` captures the specification fed in by the user. I guess you're suggesting that we ... expand it, so that it would accumulate information as we acquire it from the commit messages in the repository?

(or maybe I misunderstand what kind of connection you are asking for here? Do you want the information stored inline, or some sort of hashmap?)

Anyway I do not think we need to block this PR on this particular suggested changes. We can explore this question after this lands.

Reply:
> Anyway I do not think we need to block this PR on this particular suggested changes. We can explore this question after this lands.

Definitely agree. Just a comment now that you have additional data stored in a data structure from the GH API pulls.  It would be useful IMO for these data to move through the testing execution and be available at the final report generation stage.  I think that we lose the `Commit` struct early in the path and the `Bound` data structure is what is available for report generation.  My convoluted comment above was simply wondering out loud if there is a way to connect those pieces so that we have the commit messages for use in the report.  Maybe this is not necessary / is low priority?  GH shows them in popups when you hover over a commit SHA and we *do* have that in the report.

Reply:
Okay, so lets attack this in a separate follow-up task.

Reply:
Filed as https://github.com/rust-lang/cargo-bisect-rustc/issues/64